File: 240116/비오는 날/rainy-day.py
n = int(input())

class rain:

    def __init__(self,data,day,weather):
        self.data = data
        self.day = day 
        self.weather = weather
arr = []
for i in range(n):
    data,day,weather = map(str,input().split())
    if weather == "Rain":
        arr.append(rain(data,day,weather))

min_val = 0

for i in range(1,len(arr)):
    if arr[i].data < arr[min_val].data:
        min_val = i 
print(arr[min_val].data,end=" ")
print(arr[min_val].day,end=" ")
print(arr[min_val].weather)

File: 240116/비오는 날/rainy-day.java
import java.util.Scanner;
class oneday{
    String time;
    String day;
    String weather;

    public oneday(String time, String day, String weather){
        this.time = time;
        this.day = day;
        this.weather = weather;
    }
}
public class Main {
    public static oneday ans = new oneday("9999-99-99", "", "");
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        
        
       
        for(int i = 0 ; i<n; i++){
            String time = sc.next();
            String day = sc.next();
            String weather = sc.next();
            oneday f = new oneday(time,day,weather);
            if (weather.equals("Rain")){
                
                if (ans.time.compareTo(f.time)>0){
                    ans = f;
                }
            }
        }

        
        System.out.print(ans.time+" "+ans.day+" "+ans.weather);
    }
}